### PR TITLE
[SchemaRegistry] final api review changes

### PR DESCRIPTION
**Pull request by swathipil**

### CODE REVIEW

File: sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,13 @@
 
 ### Breaking Changes
 
+- `version` instance variable in `SchemaProperties` has been removed.  
+- `schema_definition` instance variable in `Schema` has been renamed `definition`.
+- `id` parameter in `get_schema` method on sync and async `SchemaRegistryClient` has been renamed `id`.
```
Comment (yunhaoling) [inline sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md:21]: should be id to schema_id?

```suggestion
- `id` parameter in `get_schema` method on sync and async `SchemaRegistryClient` has been renamed `schema_id`.
```

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_common/_constants.py [hunk 2]
```diff
@@ -33,3 +33,5 @@ class SchemaFormat(str, Enum):
 
     AVRO = "avro"
     """Represents the Apache Avro schema format."""
+
+SUPPORTED_API_VERSIONS = ["2021-10"]
```
Comment (yunhaoling) [inline sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_common/_constants.py:37]: check here for api version:

https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/search/azure-search-documents/azure/search/documents/_api_versions.py

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py [hunk 3]
```diff
@@ -62,6 +65,14 @@ class SchemaRegistryClient(object):
...
     def __init__(self, fully_qualified_namespace, credential, **kwargs):
         # type: (str, TokenCredential, Any) -> None
+        api_version = kwargs.get("api_version")
+        if api_version and api_version not in SUPPORTED_API_VERSIONS:
+            versions = "\n".join(SUPPORTED_API_VERSIONS)
+            raise ValueError(
+                "Unsupported API version '{}'. Please select from:\n{}".format(
+                    api_version, versions
+                )
+            )
```
Comment (yunhaoling) [inline sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py:75]: we don't handle api version at our application level, we pass it to the underlying azure-core

you could also refer to the search for api_version:

https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/search/azure-search-documents/azure/search/documents/_search_client.py#L65-L105

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/aio/_schema_registry_client_async.py [hunk 4]
```diff
@@ -65,6 +68,14 @@ def __init__(
...
         **kwargs: Any
     ) -> None:
+        api_version = kwargs.get("api_version")
+        if api_version and api_version not in SUPPORTED_API_VERSIONS:
+            versions = "\n".join(SUPPORTED_API_VERSIONS)
+            raise ValueError(
+                "Unsupported API version '{}'. Please select from:\n{}".format(
+                    api_version, versions
+                )
+            )
```
Comment (yunhaoling) [inline sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/aio/_schema_registry_client_async.py:78]: same as the sync comment

File: sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md (hunk 1 above)
Comment (swathipil) [reply sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md:21]: good catch thanks!

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_common/_constants.py [hunk 5]
```diff
@@ -33,3 +33,14 @@ class SchemaFormat(str, Enum):
 
     AVRO = "avro"
     """Represents the Apache Avro schema format."""
+
+class ApiVersion(str, Enum):
+    """
+    Represents the Schema Registry API version to use for requests.
+    """
+
+    V2020_10 = "2020-10"
```
Comment (yunhaoling) [inline sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_common/_constants.py:42]: should be 2021?
```suggestion
    V2021_10 = "2021-10"
```

File: sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md [hunk 6]
```diff
@@ -13,9 +13,16 @@
 - `SchemaProperties` has the following instance variables: `id`, `format`, `version`.
 - `Schema` has the following properties: `properties` and `schema_definition`.
 - `SchemaFormat` provides the schema format to be stored by the service. Currently, the only supported format is `Avro`.
+- `api_version` has been added as a keyword arg to the sync and async `SchemaRegistryClient` constructors:
+  - currently defaults to and only accepts the string `"2021-10"`
```
Comment (yunhaoling) [inline sdk/schemaregistry/azure-schemaregistry/CHANGELOG.md:17]: I probably would not highlight it in the changelog, docstring should be enough..

```suggestion
- `api_version` has been added as a keyword arg to the sync and async `SchemaRegistryClient` constructors.
```

File: sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_common/_constants.py (hunk 5 above)
Comment (swathipil) [reply sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_common/_constants.py:42]: yep 🤦

**Review by yunhaoling — APPROVED** [review]

**Pull request merged**